File: solutions/kotlin/ChocolateFeast.kt
import kotlin.math.floor

// Complete the chocolateFeast function below.
fun chocolateFeast(n: Int, c: Int, m: Int): Int {
    var wrappers: Int = floor(n/c.toDouble()).toInt()
    var ate: Int = wrappers

    var aditPieces: Int
    var remain: Int

    while(wrappers >= m){
        aditPieces = floor(wrappers / m.toDouble()).toInt()
        remain = wrappers - (aditPieces * m)

        ate += aditPieces
        remain += aditPieces
        wrappers = remain
    }

    return ate
}

File: solutions/ruby/chocolate_feast.rb
def chocolate_feast n, c, m

	wrappers = (n/c).floor
	ate = wrappers

	while wrappers >= m
		adit_pieces = (wrappers / m).floor

		remain = wrappers - adit_pieces * m

		ate = ate + adit_pieces

		remain = remain + (wrappers / m).floor

		wrappers = remain
	end
	
	ate
end